src/native/event_stream_rpc_client.c

```diff
@@ -221,7 +221,7 @@ jint JNICALL Java_software_amazon_awssdk_crt_eventstream_ClientConnection_client
         .socket_options = socket_options,
         .tls_options = conn_options_ptr,
         .user_data = callback_data,
-        .port = (uint16_t)port,
+        .port = (uint32_t)port,
```

Comment:
I don't think this is safe since jint can be negative.

Reply:
It's safe in that -1 becomes 0xFFFFFFFF
The [VSOCK docs](https://man.archlinux.org/man/vsock.7.en#Address_format) even use language like:
> There are several special addresses: VMADDR_CID_ANY (-1U) means any address for binding

which reads to me like "32 bits signed unsigned whatever"

Reply:
It's not safe if it's done in a native layer unknown to the user.

Reply:
When I google "vsock java", the first hit that comes up is vsockj, which uses `int` for port and `-1` for the ANY port
https://github.com/Cloud-Architects/vsockj/blob/487ebc0c61281bb109a2801ed19a85d3ab653395/vsockj-core/src/main/java/solutions/cloudarchitects/vsockj/VSockAddress.java#L26

Reply:
The 2nd hit on google also uses int
https://github.com/kohlschutter/junixsocket/blob/cdb4c1a150222f88e62e025aada742ec3ccabcab/junixsocket-common/src/main/java/org/newsclub/net/unix/AFVSOCKSocketAddress.java#L94

Reply:
I'm guessing it also plays well with Java's [URI](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#getPort--) class which uses int

Reply:
Documented:
> For 32bit values exceeding Integer.MAX_VALUE use two's complement (i.e. -1 == 0xFFFFFFFF).